File: mobile/src/screens/HomeScreen.js
import React, { useState, useEffect } from 'react';
import { View, Text, FlatList, StyleSheet, ActivityIndicator, TouchableOpacity } from 'react-native';
import AsyncStorage from '@react-native-async-storage/async-storage';
import { hostelService } from '../services/api';

const HomeScreen = ({ navigation }) => {
  const [hostels, setHostels] = useState([]);
  const [loading, setLoading] = useState(true);
  const [user, setUser] = useState(null);

  useEffect(() => {
    loadUserAndHostels();
  }, []);

  const loadUserAndHostels = async () => {
    try {
      const userData = await AsyncStorage.getItem('user');
      setUser(JSON.parse(userData));

      if (JSON.parse(userData)?.isSubscribed) {
        const response = await hostelService.getAllHostels();
        setHostels(response.data);
      }
    } catch (error) {
      console.error('Error loading data:', error);
    } finally {
      setLoading(false);
    }
  };

  if (!user?.isSubscribed) {
    return (
      <View style={styles.subscriptionContainer}>
        <Text style={styles.title}>Unlock SMARTSTAY CHUKA</Text>
        <Text style={styles.subtitle}>Subscribe to view all hostels</Text>
        <Text style={styles.price}>KES 253 for 30 days</Text>
        <TouchableOpacity
          style={styles.subscribeButton}
          onPress={() => navigation.navigate('Payment')}
        >
          <Text style={styles.subscribeButtonText}>Subscribe Now</Text>
        </TouchableOpacity>
      </View>
    );
  }

  if (loading) {
    return (
      <View style={styles.centerContainer}>
        <ActivityIndicator size="large" color="#667eea" />
      </View>
    );
  }

  return (
    <View style={styles.container}>
      <Text style={styles.title}>Find Your Perfect Hostel</Text>
      <FlatList
        data={hostels}
        keyExtractor={(item) => item._id}
        renderItem={({ item }) => (
          <TouchableOpacity
            style={styles.hostelCard}
            onPress={() => navigation.navigate('HostelDetail', { id: item._id })}
          >
            <Text style={styles.hostelName}>{item.name}</Text>
            <Text style={styles.hostelText}>📍 {item.location}</Text>
            <Text style={styles.hostelText}>📞 {item.phoneNumber}</Text>
            <Text style={styles.hostelDescription}>{item.description}</Text>
          </TouchableOpacity>
        )}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#f5f5f5',
    padding: 15,
  },
  centerContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  subscriptionContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#f5f5f5',
  },
  title: {
    fontSize: 22,
    fontWeight: 'bold',
    color: '#333',
    marginBottom: 15,
  },
  subtitle: {
    fontSize: 16,
    color: '#666',
    textAlign: 'center',
    marginBottom: 20,
  },
  price: {
    fontSize: 28,
    fontWeight: 'bold',
    color: '#667eea',
    marginBottom: 30,
  },
  subscribeButton: {
    backgroundColor: '#667eea',
    paddingVertical: 14,
    paddingHorizontal: 40,
    borderRadius: 8,
  },
  subscribeButtonText: {
    color: 'white',
    fontSize: 16,
    fontWeight: '600',
  },
  hostelCard: {
    backgroundColor: 'white',
    padding: 15,
    borderRadius: 8,
    marginBottom: 10,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.1,
    shadowRadius: 4,
    elevation: 3,
  },
  hostelName: {
    fontSize: 18,
    fontWeight: '600',
    color: '#333',
    marginBottom: 8,
  },
  hostelText: {
    fontSize: 14,
    color: '#666',
    marginBottom: 5,
  },
  hostelDescription: {
    fontSize: 13,
    color: '#777',
    marginTop: 8,
    lineHeight: 18,
  },
});

export default HomeScreen;

File: mobile/lib/screens/home_screen.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import '../services/auth_service.dart';
import '../services/hostel_service.dart';

class HomeScreen extends StatefulWidget {
  const HomeScreen({Key? key}) : super(key: key);

  @override
  State<HomeScreen> createState() => _HomeScreenState();
}

class _HomeScreenState extends State<HomeScreen> {
  @override
  void initState() {
    super.initState();
    WidgetsBinding.instance.addPostFrameCallback((_) {
      context.read<HostelService>().fetchHostels();
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('Chuka Hostels'),
        elevation: 0,
        backgroundColor: const Color(0xFF1A237E),
        foregroundColor: Colors.white,
        actions: [
          IconButton(
            icon: const Icon(Icons.person),
            onPressed: () => Navigator.of(context).pushNamed('/profile'),
          ),
        ],
      ),
      body: Consumer2<AuthService, HostelService>(
        builder: (context, authService, hostelService, _) {
          if (!authService.currentUser!.isSubscribed) {
            return Center(
              child: Column(
                mainAxisAlignment: MainAxisAlignment.center,
                children: [
                  const Icon(
                    Icons.lock,
                    size: 64,
                    color: Color(0xFF1A237E),
                  ),
                  const SizedBox(height: 16),
                  const Text(
                    'Subscribe to View Hostels',
                    style: TextStyle(
                      fontSize: 20,
                      fontWeight: FontWeight.bold,
                    ),
                  ),
                  const SizedBox(height: 8),
                  Text(
                    'Unlock full access for KES 263',
                    style: Theme.of(context).textTheme.bodyMedium,
                  ),
                  const SizedBox(height: 24),
                  ElevatedButton(
                    onPressed: () =>
                        Navigator.of(context).pushNamed('/subscription'),
                    child: const Text('Subscribe Now'),
                  ),
                ],
              ),
            );
          }

          if (hostelService.isLoading) {
            return const Center(child: CircularProgressIndicator());
          }

          return RefreshIndicator(
            onRefresh: () => hostelService.fetchHostels(),
            child: ListView.builder(
              itemCount: hostelService.hostels.length,
              itemBuilder: (context, index) {
                final hostel = hostelService.hostels[index];
                return GestureDetector(
                  onTap: () => Navigator.of(context)
                      .pushNamed('/hostel-detail', arguments: hostel.id),
                  child: Card(
                    margin: const EdgeInsets.all(12),
                    child: Column(
                      crossAxisAlignment: CrossAxisAlignment.start,
                      children: [
                        ClipRRect(
                          borderRadius: const BorderRadius.only(
                            topLeft: Radius.circular(12),
                            topRight: Radius.circular(12),
                          ),
                          child: Image.network(
                            hostel.image,
                            height: 200,
                            width: double.infinity,
                            fit: BoxFit.cover,
                            errorBuilder: (context, error, stackTrace) =>
                                Container(
                              height: 200,
                              color: Colors.grey[300],
                              child: const Icon(Icons.image_not_supported),
                            ),
                          ),
                        ),
                        Padding(
                          padding: const EdgeInsets.all(16),
                          child: Column(
                            crossAxisAlignment: CrossAxisAlignment.start,
                            children: [
                              Text(
                                hostel.name,
                                style: const TextStyle(
                                  fontSize: 18,
                                  fontWeight: FontWeight.bold,
                                ),
                              ),
                              const SizedBox(height: 8),
                              Row(
                                children: [
                                  const Icon(Icons.location_on_outlined,
                                      size: 16),
                                  const SizedBox(width: 4),
                                  Expanded(child: Text(hostel.location)),
                                ],
                              ),
                              const SizedBox(height: 8),
                              Wrap(
                                spacing: 8,
                                children: hostel.amenities
                                    .take(3)
                                    .map((amenity) => Chip(
                                          label: Text(amenity),
                                          backgroundColor:
                                              const Color(0xFF009688)
                                                  .withOpacity(0.2),
                                        ))
                                    .toList(),
                              ),
                            ],
                          ),
                        ),
                      ],
                    ),
                  ),
                );
              },
            ),
          );
        },
      ),
    );
  }
}
